docs/content/config/advanced/kubernetes.md

```diff
@@ -2,392 +2,374 @@
+  externalTrafficPolicy: Local
+
+  selector:
+    app: mailserver
+
+  ports:
+    # Transfer
+    - name: transfer
+      port: 25
+      targetPort: transfer
+      protocol: TCP
+    # ESMTP with implicit TLS
+    - name: esmtp-implicit
+      port: 465
+      targetPort: esmtp-implicit
+      protocol: TCP
+    # ESMTP with explicit TLS (STARTTLS)
+    - name: esmtp-explicit
+      port: 587
+      targetPort: esmtp-explicit
+      protocol: TCP
+    # IMAPS with implicit TLS
+    - name: imap-implicit
+      port: 993
+      targetPort: imap-implicit
+      protocol: TCP
+
+```
 
-The simplest way is to expose mailserver as a [Service][k8s-network-service] with [external IPs][k8s-network-external-ip].
+### Deployments
 
-!!! example
+Last but not least, the `Deployment` becomes the most complex component. The deployment brings maximal security measures without compromising on ease of use.
 
-    ```yaml
-    kind: Service
-    apiVersion: v1
+```yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+
+metadata:
+  name: mailserver
+
+  annotations:
+    ignore-check.kube-linter.io/run-as-non-root: >-
+      The mailserver needs to run as root
+    ignore-check.kube-linter.io/privileged-ports: >-
+      The mailserver needs privilegdes ports
+    ignore-check.kube-linter.io/no-read-only-root-fs: >-
+      There are too many files written to make The
+      root FS read-only
+
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: mailserver
+
+  template:
     metadata:
-      name: mailserver
       labels:
         app: mailserver
+
+      annotations:
+        container.apparmor.security.beta.kubernetes.io/mailserver: runtime/default
+
     spec:
-      selector:
-        app: mailserver
-      ports:
-        - name: smtp
-          port: 25
-          targetPort: smtp
+      hostname: mailserver
+      containers:
+        - name: mailserver
+          image: ghcr.io/docker-mailserver/docker-mailserver:10.0.0
+          imagePullPolicy: IfNotPresent
+
+          securityContext:
+            allowPrivilegeEscalation: false
+            readOnlyRootFilesystem: false
+            runAsUser: 0
+            runAsGroup: 0
+            runAsNonRoot: false
+            privileged: false
+            capabilities:
+              add:
+                # file permission capabilities
+                - CHOWN
+                - FOWNER
+                - MKNOD
+                - SETGID
+                - SETUID
+                - DAC_OVERRIDE
+                # network capabilities
+                - NET_ADMIN
+                - NET_BIND_SERVICE
+                # miscellaneous  capabilities
+                - SYS_CHROOT
+                - SYS_PTRACE
+                - KILL
+              drop: [ALL]
+            seccompProfile:
+              type: RuntimeDefault
+
+          resources:
+            limits:
+              memory: 4Gi
+              cpu: 1500m
+            requests:
+              memory: 2Gi
+              cpu: 600m
```

Comment:
We might want to add comments that these values can/should be tuned based on the scale of the implementation. I run the below for 2 (real) users, ClamAV off, and ~400 emails a day. Spam sucks.
```
            requests:
              memory: "500Mi"
              cpu: "750m"
            limits:
              memory: "1500Mi"
              cpu: "900m"
```

I mention this because K8s will start refusing to schedule pods if you exceed your memory/cpu budgets, and you don't _always_ need the suggested amounts. Users should be aware that they can tune this.